<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from jaco.xacro                     | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="jaco" xmlns:body="http://playerstage.sourceforge.net/gazebo/xmlschema/#body" xmlns:controller="http://playerstage.sourceforge.net/gazebo/xmlschema/#controller" xmlns:gazebo="http://playerstage.sourceforge.net/gazebo/xmlschema/#gz" xmlns:geom="http://playerstage.sourceforge.net/gazebo/xmlschema/#geom" xmlns:interface="http://playerstage.sourceforge.net/gazebo/xmlschema/#interface" xmlns:joint="http://playerstage.sourceforge.net/gazebo/xmlschema/#joint" xmlns:model="http://playerstage.sourceforge.net/gazebo/xmlschema/#model" xmlns:physics="http://playerstage.sourceforge.net/gazebo/xmlschema/#physics" xmlns:renderable="http://playerstage.sourceforge.net/gazebo/xmlschema/#renderable" xmlns:rendering="http://playerstage.sourceforge.net/gazebo/xmlschema/#rendering" xmlns:sensor="http://playerstage.sourceforge.net/gazebo/xmlschema/#sensor" xmlns:xacro="http://ros.org/wiki/xacro" xmlns:xi="http://www.w3.org/2001/XInclude">
  <!--prefix means arm name, such as jaco/jaco_left, which is corresponding to jaco.yaml/jaco_left.yaml-->
  <link name="world_frame">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0 0 0"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0 0 0"/>
      </geometry>
    </collision>
  </link>
  <joint name="support_to_base" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="world_frame"/>
    <child link="support_link"/>
    <axis xyz="0 0 1"/>
  </joint>
  <link name="support_link">
    <inertial>
      <origin rpy="0 0 0" xyz="0.029769 -0.0126599999999999 0.97862"/>
      <mass value="8.1734"/>
      <inertia ixx="3.14464667119818" ixy="0.0415389708201151" ixz="-0.00168971152269903" iyy="-0.0861920217938598" iyz="-3.01021801246017" izz="0.08619202179386"/>
    </inertial>
    <visual>
      <origin rpy="1.5707963267949 0 -1.5707963267949" xyz="0 0 -0.28"/>
      <geometry>
        <mesh filename="package://move_group/urdf/support/meshes/support.STL"/>
      </geometry>
      <material name="">
        <color rgba="1 1 1 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="1.5707963267949 0 -1.5707963267949" xyz="0 0 -0.28"/>
      <geometry>
        <mesh filename="package://move_group/urdf/support/meshes/support.STL"/>
      </geometry>
    </collision>
  </link>
  <gazebo>
    <plugin filename="libgazebo_ros_control.so" name="gazebo_ros_controller_manager">
      <!-- <robotNamespace>/</robotNamespace> -->
      <robotParam>robot_description</robotParam>
    </plugin>
  </gazebo>
  <link name="jaco_link_base">
    <inertial>
      <mass value="1"/>
      <inertia ixx="0.1" ixy="0" ixz="0" iyy="0.1" iyz="0" izz="0.1"/>
      <origin/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="package://move_group/urdf/jaco_description/meshes/jaco_link_base.min.dae" scale="0.0254 0.0254 0.0254"/>
      </geometry>
      <material name="carbon_fiber">
        <color rgba="0.792156862745098 0.819607843137255 0.933333333333333 1"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://move_group/urdf/jaco_description/meshes/jaco_link_base.stl" scale="0.0254 0.0254 0.0254"/>
      </geometry>
    </collision>
  </link>
  <joint name="jaco_joint_base" type="fixed">
    <parent link="support_link"/>
    <child link="jaco_link_base"/>
    <axis xyz="0 0 0"/>
    <origin rpy="3.141592653 -0.785 1.5707963267949" xyz="0 0 1.22"/>
    <limit effort="30" lower="-6.28318530718" upper="6.28318530718" velocity="5"/>
  </joint>
  <link name="jaco_link_1">
    <inertial>
      <mass value="1"/>
      <inertia ixx="0.1" ixy="0" ixz="0" iyy="0.1" iyz="0" izz="0.1"/>
      <origin/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="package://move_group/urdf/jaco_description/meshes/jaco_link_1.min.dae" scale="0.0254 0.0254 0.0254"/>
      </geometry>
      <material name="carbon_fiber">
        <color rgba="0.792156862745098 0.819607843137255 0.933333333333333 1"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://move_group/urdf/jaco_description/meshes/jaco_link_1.stl" scale="0.0254 0.0254 0.0254"/>
      </geometry>
    </collision>
  </link>
  <joint name="jaco_joint_1" type="revolute">
    <parent link="jaco_link_base"/>
    <child link="jaco_link_1"/>
    <axis xyz="0 0 1"/>
    <origin rpy="3.14159265359 0 0" xyz="0 0 0.1535"/>
    <limit effort="30" lower="-6.28318530718" upper="6.28318530718" velocity="5"/>
  </joint>
  <transmission name="jaco_joint_1_transmission">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="jaco_joint_1">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="jaco_joint_1_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <link name="jaco_link_2">
    <inertial>
      <mass value="1"/>
      <inertia ixx="0.1" ixy="0" ixz="0" iyy="0.1" iyz="0" izz="0.1"/>
      <origin/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="package://move_group/urdf/jaco_description/meshes/jaco_link_2.min.dae" scale="0.0254 0.0254 0.0254"/>
      </geometry>
      <material name="carbon_fiber">
        <color rgba="0.792156862745098 0.819607843137255 0.933333333333333 1"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://move_group/urdf/jaco_description/meshes/jaco_link_2.stl" scale="0.0254 0.0254 0.0254"/>
      </geometry>
    </collision>
  </link>
  <joint name="jaco_joint_2" type="revolute">
    <parent link="jaco_link_1"/>
    <child link="jaco_link_2"/>
    <axis xyz="0 0 1"/>
    <origin rpy="-1.57079632679 -1.57079632679 0" xyz="0 0 -0.1185"/>
    <limit effort="30" lower="0" upper="6.28318530718" velocity="5"/>
  </joint>
  <transmission name="jaco_joint_2_transmission">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="jaco_joint_2">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="jaco_joint_2_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <link name="jaco_link_3">
    <inertial>
      <mass value="1"/>
      <inertia ixx="0.1" ixy="0" ixz="0" iyy="0.1" iyz="0" izz="0.1"/>
      <origin/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="package://move_group/urdf/jaco_description/meshes/jaco_link_3.min.dae" scale="0.0254 0.0254 0.0254"/>
      </geometry>
      <material name="carbon_fiber">
        <color rgba="0.792156862745098 0.819607843137255 0.933333333333333 1"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://move_group/urdf/jaco_description/meshes/jaco_link_3.stl" scale="0.0254 0.0254 0.0254"/>
      </geometry>
    </collision>
  </link>
  <joint name="jaco_joint_3" type="revolute">
    <parent link="jaco_link_2"/>
    <child link="jaco_link_3"/>
    <axis xyz="0 0 1"/>
    <origin rpy="3.14159265359 0 3.14159265359" xyz="0.41 0 0"/>
    <limit effort="30" lower="0" upper="6.28318530718" velocity="5"/>
  </joint>
  <transmission name="jaco_joint_3_transmission">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="jaco_joint_3">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="jaco_joint_3_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <link name="jaco_link_4">
    <inertial>
      <mass value="0.5"/>
      <inertia ixx="0.01" ixy="0" ixz="0" iyy="0.01" iyz="0" izz="0.01"/>
      <origin/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="package://move_group/urdf/jaco_description/meshes/jaco_link_4_5.min.dae" scale="0.0254 0.0254 0.0254"/>
      </geometry>
      <material name="carbon_fiber">
        <color rgba="0.792156862745098 0.819607843137255 0.933333333333333 1"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://move_group/urdf/jaco_description/meshes/jaco_link_4_5.stl" scale="0.0254 0.0254 0.0254"/>
      </geometry>
    </collision>
  </link>
  <joint name="jaco_joint_4" type="revolute">
    <parent link="jaco_link_3"/>
    <child link="jaco_link_4"/>
    <axis xyz="0 0 1"/>
    <origin rpy="0 -1.57079632679 0" xyz="0.207 0 -0.01125"/>
    <limit effort="30" lower="-6.28318530718" upper="6.28318530718" velocity="5"/>
  </joint>
  <transmission name="jaco_joint_4_transmission">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="jaco_joint_4">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="jaco_joint_4_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <link name="jaco_link_5">
    <inertial>
      <mass value="0.5"/>
      <inertia ixx="0.01" ixy="0" ixz="0" iyy="0.01" iyz="0" izz="0.01"/>
      <origin/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="package://move_group/urdf/jaco_description/meshes/jaco_link_4_5.min.dae" scale="0.0254 0.0254 0.0254"/>
      </geometry>
      <material name="carbon_fiber">
        <color rgba="0.792156862745098 0.819607843137255 0.933333333333333 1"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://move_group/urdf/jaco_description/meshes/jaco_link_4_5.stl" scale="0.0254 0.0254 0.0254"/>
      </geometry>
    </collision>
  </link>
  <joint name="jaco_joint_5" type="revolute">
    <parent link="jaco_link_4"/>
    <child link="jaco_link_5"/>
    <axis xyz="0 0 1"/>
    <origin rpy="0 0.959931088597 3.14159265359" xyz="0.0343 0 -0.06588"/>
    <limit effort="30" lower="-6.28318530718" upper="6.28318530718" velocity="5"/>
  </joint>
  <transmission name="jaco_joint_5_transmission">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="jaco_joint_5">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="jaco_joint_5_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <link name="jaco_link_hand">
    <inertial>
      <mass value="1"/>
      <inertia ixx="0.1" ixy="0" ixz="0" iyy="0.1" iyz="0" izz="0.1"/>
      <origin/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="package://move_group/urdf/jaco_description/meshes/jaco_link_hand.min.dae" scale="0.0254 0.0254 0.0254"/>
      </geometry>
      <material name="carbon_fiber">
        <color rgba="0.792156862745098 0.819607843137255 0.933333333333333 1"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://move_group/urdf/jaco_description/meshes/jaco_link_hand.stl" scale="0.0254 0.0254 0.0254"/>
      </geometry>
    </collision>
  </link>
  <joint name="jaco_joint_6" type="revolute">
    <parent link="jaco_link_5"/>
    <child link="jaco_link_hand"/>
    <axis xyz="0 0 1"/>
    <origin rpy="0 0.959931088597 3.14159265359" xyz="0.0343 0 -0.06588"/>
    <limit effort="30" lower="-6.28318530718" upper="6.28318530718" velocity="5"/>
  </joint>
  <transmission name="jaco_joint_6_transmission">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="jaco_joint_6">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="jaco_joint_6_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <link name="jaco_large_ring_cover_1">
    <inertial>
      <mass value="0"/>
      <inertia ixx="0" ixy="0" ixz="0" iyy="0" iyz="0" izz="0"/>
      <origin/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="package://move_group/urdf/jaco_description/meshes/jaco_large_ring_cover.min.dae" scale="0.0254 0.0254 0.0254"/>
      </geometry>
      <material name="carbon_fiber">
        <color rgba="0.792156862745098 0.819607843137255 0.933333333333333 1"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://move_group/urdf/jaco_description/meshes/jaco_large_ring_cover.stl" scale="0.0254 0.0254 0.0254"/>
      </geometry>
    </collision>
  </link>
  <joint name="jaco_large_ring_cover_1_joint" type="fixed">
    <parent link="jaco_link_base"/>
    <child link="jaco_large_ring_cover_1"/>
    <axis xyz="0 0 1"/>
    <origin rpy="3.14159265359 0 0" xyz="0 0 0.1535"/>
    <limit effort="30" lower="-6.28318530718" upper="6.28318530718" velocity="5"/>
  </joint>
  <link name="jaco_large_ring_cover_2">
    <inertial>
      <mass value="0"/>
      <inertia ixx="0" ixy="0" ixz="0" iyy="0" iyz="0" izz="0"/>
      <origin/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="package://move_group/urdf/jaco_description/meshes/jaco_large_ring_cover.min.dae" scale="0.0254 0.0254 0.0254"/>
      </geometry>
      <material name="carbon_fiber">
        <color rgba="0.792156862745098 0.819607843137255 0.933333333333333 1"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://move_group/urdf/jaco_description/meshes/jaco_large_ring_cover.stl" scale="0.0254 0.0254 0.0254"/>
      </geometry>
    </collision>
  </link>
  <joint name="jaco_large_ring_cover_2_joint" type="fixed">
    <parent link="jaco_link_1"/>
    <child link="jaco_large_ring_cover_2"/>
    <axis xyz="0 0 1"/>
    <origin rpy="-1.57079632679 -1.57079632679 0" xyz="0 0 -0.1185"/>
    <limit effort="30" lower="0" upper="6.28318530718" velocity="5"/>
  </joint>
  <link name="jaco_large_ring_cover_3">
    <inertial>
      <mass value="0"/>
      <inertia ixx="0" ixy="0" ixz="0" iyy="0" iyz="0" izz="0"/>
      <origin/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="package://move_group/urdf/jaco_description/meshes/jaco_large_ring_cover.min.dae" scale="0.0254 0.0254 0.0254"/>
      </geometry>
      <material name="carbon_fiber">
        <color rgba="0.792156862745098 0.819607843137255 0.933333333333333 1"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://move_group/urdf/jaco_description/meshes/jaco_large_ring_cover.stl" scale="0.0254 0.0254 0.0254"/>
      </geometry>
    </collision>
  </link>
  <joint name="jaco_large_ring_cover_3_joint" type="fixed">
    <parent link="jaco_link_2"/>
    <child link="jaco_large_ring_cover_3"/>
    <axis xyz="0 0 1"/>
    <origin rpy="3.14159265359 0 3.14159265359" xyz="0.41 0 0"/>
    <limit effort="30" lower="0" upper="6.28318530718" velocity="5"/>
  </joint>
  <link name="jaco_small_ring_cover_1">
    <inertial>
      <mass value="0"/>
      <inertia ixx="0" ixy="0" ixz="0" iyy="0" iyz="0" izz="0"/>
      <origin/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="package://move_group/urdf/jaco_description/meshes/jaco_small_ring_cover.min.dae" scale="0.0254 0.0254 0.0254"/>
      </geometry>
      <material name="carbon_fiber">
        <color rgba="0.792156862745098 0.819607843137255 0.933333333333333 1"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://move_group/urdf/jaco_description/meshes/jaco_small_ring_cover.stl" scale="0.0254 0.0254 0.0254"/>
      </geometry>
    </collision>
  </link>
  <joint name="jaco_large_ring_cover_4_joint" type="fixed">
    <parent link="jaco_link_3"/>
    <child link="jaco_small_ring_cover_1"/>
    <axis xyz="0 0 1"/>
    <origin rpy="0 -1.57079632679 0" xyz="0.207 0 -0.01125"/>
    <limit effort="30" lower="-6.28318530718" upper="6.28318530718" velocity="5"/>
  </joint>
  <link name="jaco_small_ring_cover_2">
    <inertial>
      <mass value="0"/>
      <inertia ixx="0" ixy="0" ixz="0" iyy="0" iyz="0" izz="0"/>
      <origin/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="package://move_group/urdf/jaco_description/meshes/jaco_small_ring_cover.min.dae" scale="0.0254 0.0254 0.0254"/>
      </geometry>
      <material name="carbon_fiber">
        <color rgba="0.792156862745098 0.819607843137255 0.933333333333333 1"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://move_group/urdf/jaco_description/meshes/jaco_small_ring_cover.stl" scale="0.0254 0.0254 0.0254"/>
      </geometry>
    </collision>
  </link>
  <joint name="jaco_large_ring_cover_5_joint" type="fixed">
    <parent link="jaco_link_4"/>
    <child link="jaco_small_ring_cover_2"/>
    <axis xyz="0 0 1"/>
    <origin rpy="0 0.959931088597 3.14159265359" xyz="0.0343 0 -0.06588"/>
    <limit effort="30" lower="-6.28318530718" upper="6.28318530718" velocity="5"/>
  </joint>
  <link name="jaco_small_ring_cover_3">
    <inertial>
      <mass value="0"/>
      <inertia ixx="0" ixy="0" ixz="0" iyy="0" iyz="0" izz="0"/>
      <origin/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="package://move_group/urdf/jaco_description/meshes/jaco_small_ring_cover.min.dae" scale="0.0254 0.0254 0.0254"/>
      </geometry>
      <material name="carbon_fiber">
        <color rgba="0.792156862745098 0.819607843137255 0.933333333333333 1"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://move_group/urdf/jaco_description/meshes/jaco_small_ring_cover.stl" scale="0.0254 0.0254 0.0254"/>
      </geometry>
    </collision>
  </link>
  <joint name="jaco_large_ring_cover_6_joint" type="fixed">
    <parent link="jaco_link_5"/>
    <child link="jaco_small_ring_cover_3"/>
    <axis xyz="0 0 1"/>
    <origin rpy="0 0.959931088597 3.14159265359" xyz="0.0343 0 -0.06588"/>
    <limit effort="30" lower="-6.28318530718" upper="6.28318530718" velocity="5"/>
  </joint>
  <link name="jaco_link_finger_1">
    <inertial>
      <mass value="0.25"/>
      <inertia ixx="0.01" ixy="0" ixz="0" iyy="0.01" iyz="0" izz="0.01"/>
      <origin/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="package://move_group/urdf/jaco_description/meshes/jaco_link_finger.min.dae" scale="0.0254 0.0254 0.0254"/>
      </geometry>
      <material name="carbon_fiber">
        <color rgba="0.792156862745098 0.819607843137255 0.933333333333333 1"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://move_group/urdf/jaco_description/meshes/jaco_link_finger.stl" scale="0.0254 0.0254 0.0254"/>
      </geometry>
    </collision>
  </link>
  <joint name="jaco_joint_finger_1" type="revolute">
    <parent link="jaco_link_hand"/>
    <child link="jaco_link_finger_1"/>
    <axis xyz="0 0 1"/>
    <origin rpy="-1.7983 1.117 3.1416" xyz="-0.032578 0.0033 -0.12131"/>
    <limit effort="30" lower="0" upper="0.872664625997" velocity="5"/>
  </joint>
  <transmission name="jaco_joint_finger_1_transmission">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="jaco_joint_finger_1">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="jaco_joint_finger_1_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <link name="jaco_link_finger_2">
    <inertial>
      <mass value="0.25"/>
      <inertia ixx="0.01" ixy="0" ixz="0" iyy="0.01" iyz="0" izz="0.01"/>
      <origin/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="package://move_group/urdf/jaco_description/meshes/jaco_link_finger.min.dae" scale="0.0254 0.0254 0.0254"/>
      </geometry>
      <material name="carbon_fiber">
        <color rgba="0.792156862745098 0.819607843137255 0.933333333333333 1"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://move_group/urdf/jaco_description/meshes/jaco_link_finger.stl" scale="0.0254 0.0254 0.0254"/>
      </geometry>
    </collision>
  </link>
  <joint name="jaco_joint_finger_2" type="revolute">
    <parent link="jaco_link_hand"/>
    <child link="jaco_link_finger_2"/>
    <axis xyz="0 0 1"/>
    <origin rpy="-1.5161 1.1459 0.23978" xyz="0.027788 0.0206 -0.12131"/>
    <limit effort="30" lower="0" upper="0.872664625997" velocity="5"/>
  </joint>
  <transmission name="jaco_joint_finger_2_transmission">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="jaco_joint_finger_2">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="jaco_joint_finger_2_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <link name="jaco_link_finger_3">
    <inertial>
      <mass value="0.25"/>
      <inertia ixx="0.01" ixy="0" ixz="0" iyy="0.01" iyz="0" izz="0.01"/>
      <origin/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="package://move_group/urdf/jaco_description/meshes/jaco_link_finger.min.dae" scale="0.0254 0.0254 0.0254"/>
      </geometry>
      <material name="carbon_fiber">
        <color rgba="0.792156862745098 0.819607843137255 0.933333333333333 1"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://move_group/urdf/jaco_description/meshes/jaco_link_finger.stl" scale="0.0254 0.0254 0.0254"/>
      </geometry>
    </collision>
  </link>
  <joint name="jaco_joint_finger_3" type="revolute">
    <parent link="jaco_link_hand"/>
    <child link="jaco_link_finger_3"/>
    <axis xyz="0 0 1"/>
    <origin rpy="-1.6222 1.117 -0.23615" xyz="0.027788 -0.0206 -0.12131"/>
    <limit effort="30" lower="0" upper="0.872664625997" velocity="5"/>
  </joint>
  <transmission name="jaco_joint_finger_3_transmission">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="jaco_joint_finger_3">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="jaco_joint_finger_3_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <link name="jaco_link_finger_tip_1">
    <inertial>
      <mass value="0.25"/>
      <inertia ixx="0.01" ixy="0" ixz="0" iyy="0.01" iyz="0" izz="0.01"/>
      <origin/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="package://move_group/urdf/jaco_description/meshes/jaco_link_finger_tip.min.dae" scale="0.0254 0.0254 0.0254"/>
      </geometry>
      <material name="carbon_fiber">
        <color rgba="0.792156862745098 0.819607843137255 0.933333333333333 1"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://move_group/urdf/jaco_description/meshes/jaco_link_finger_tip.stl" scale="0.0254 0.0254 0.0254"/>
      </geometry>
    </collision>
  </link>
  <joint name="jaco_joint_finger_tip_1" type="fixed">
    <parent link="jaco_link_finger_1"/>
    <child link="jaco_link_finger_tip_1"/>
    <axis xyz="0 0 1"/>
    <origin rpy="0 0 0" xyz="0.044094 -0.0029 0"/>
  </joint>
  <transmission name="jaco_joint_finger_tip_1_transmission">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="jaco_joint_finger_tip_1">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="jaco_joint_finger_tip_1_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <link name="jaco_link_finger_tip_2">
    <inertial>
      <mass value="0.25"/>
      <inertia ixx="0.01" ixy="0" ixz="0" iyy="0.01" iyz="0" izz="0.01"/>
      <origin/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="package://move_group/urdf/jaco_description/meshes/jaco_link_finger_tip.min.dae" scale="0.0254 0.0254 0.0254"/>
      </geometry>
      <material name="carbon_fiber">
        <color rgba="0.792156862745098 0.819607843137255 0.933333333333333 1"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://move_group/urdf/jaco_description/meshes/jaco_link_finger_tip.stl" scale="0.0254 0.0254 0.0254"/>
      </geometry>
    </collision>
  </link>
  <joint name="jaco_joint_finger_tip_2" type="fixed">
    <parent link="jaco_link_finger_2"/>
    <child link="jaco_link_finger_tip_2"/>
    <axis xyz="0 0 1"/>
    <origin rpy="0 0 0" xyz="0.044094 -0.0029 0"/>
  </joint>
  <transmission name="jaco_joint_finger_tip_2_transmission">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="jaco_joint_finger_tip_2">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="jaco_joint_finger_tip_2_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <link name="jaco_link_finger_tip_3">
    <inertial>
      <mass value="0.25"/>
      <inertia ixx="0.01" ixy="0" ixz="0" iyy="0.01" iyz="0" izz="0.01"/>
      <origin/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="package://move_group/urdf/jaco_description/meshes/jaco_link_finger_tip.min.dae" scale="0.0254 0.0254 0.0254"/>
      </geometry>
      <material name="carbon_fiber">
        <color rgba="0.792156862745098 0.819607843137255 0.933333333333333 1"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://move_group/urdf/jaco_description/meshes/jaco_link_finger_tip.stl" scale="0.0254 0.0254 0.0254"/>
      </geometry>
    </collision>
  </link>
  <joint name="jaco_joint_finger_tip_3" type="fixed">
    <parent link="jaco_link_finger_3"/>
    <child link="jaco_link_finger_tip_3"/>
    <axis xyz="0 0 1"/>
    <origin rpy="0 0 0" xyz="0.044094 -0.0029 0"/>
  </joint>
  <transmission name="jaco_joint_finger_tip_3_transmission">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="jaco_joint_finger_tip_3">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="jaco_joint_finger_tip_3_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <!-- table-->
  <joint name="world_to_table" type="fixed">
    <parent link="world_frame"/>
    <child link="table"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
  </joint>
  <link name="table">
    <visual>
      <geometry>
        <box size="4.2 1.21 0.40"/>
      </geometry>
      <origin rpy="0 0 1.5707963267949" xyz="-1.025 0 0.2"/>
      <material name="white">
        <color rgba="1 1 1 0.5"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 1.5707963267949" xyz="-1.025 0 0.2"/>
      <geometry>
        <box size="0.5 0.05 2"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="1000"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0"/>
    </inertial>
  </link>
</robot>

